applications/members/views.py

```diff
@@ -29,80 +40,92 @@ def batch(request, year):
         for item in result:
             data[row][item['branch']] = item['count']
 
-    # print(data) #prints {'B.Des': {'CSE': 1}, 'B.Tech': {'CSE': 1, 'ME': 1}} 
     return render(request, "members/year.html", {'data': data, 'year': year})
 
 
 def branch(request, programme, year, branch):
     # todo: change mail_sent to verify
     alumni = Profile.objects.filter(programme=programme, batch=year, branch=branch, verify=True)
-    # print(alumni)
     return render(request, "members/branch.html", {'data': alumni, 'batch': year, 'branch': branch})
 
-def sacbody(request):
-    return redirect('members:alumnibody')
+# def sacbody(request):
+#     return redirect('members:alumnibody')
 
 def alumnibody(request):
     return render(request, "members/alumnibody.html")
 
 
 @login_required
 def search(request):
-    key = request.GET['search']
-    profiles = Profile.objects.filter(name__icontains=key) | Profile.objects.filter(
-        roll_no__icontains=key) | Profile.objects.filter(reg_no__icontains=key)
-    if len(request.GET) > 1:
-        if request.GET['batch'] != '':
-            batch = request.GET['batch']
-            print(batch)
+    profiles = Profile.objects.all()
+    if len(request.POST) > 1:
+        if request.POST['search'] != '':
+           key = request.POST['search']
+           profiles = Profile.objects.filter(name__icontains=key) | Profile.objects.filter(
```

Comment:
We don't have to call all data again, just apply the filter on profiles, that would be fine.